argostime/crawler/crawl_utils.py

```diff
@@ -43,14 +50,14 @@ class CrawlResult():
 
     def __init__(
         self,
-        url: str=None,
-        product_name: str=None,
-        product_description: str=None,
-        product_code: str=None,
+        url: Optional[str]=None,
+        product_name: Optional[str]=None,
+        product_description: Optional[str]=None,
+        product_code: Optional[str]=None,
         normal_price: float=-1.0,
         discount_price: float=-1.0,
```

Comment:
I think the discount_price and normal_price should also be optional (since it's possible that you have neither or that you only have one of them)

Reply:
I had a quick look at this but it seems that some code depends on this not being optional. Some functions check if the value is negative or not to determine if a price is set or not.

I suggest we create a new issue and resolve this with another pull request. This one is already huge.

Reply:
Good idea, I created an issue (#58)